### PR TITLE
 #13065: Use the same action for building and running

### PR DESCRIPTION
**Pull request by dimitri-tenstorrent**
### Ticket

#13065 

### Problem description

We were working at the same time and solving similar issues from different angles resulting in very similar solutions.

1. Docker Run Action
https://github.com/tenstorrent/tt-metal/blob/cef573094c2b1ab296fc018477feabfb71576ccb/.github/actions/docker-run/action.yml#L56

2. Docker Run Action in Build Step
https://github.com/tenstorrent/tt-metal/blob/cef573094c2b1ab296fc018477feabfb71576ccb/.github/workflows/build-artifact.yaml#L100

### What's changed

The actions are actually near identical and so the generic Docker Run Action just needs to be incorporated into the build step.

Couple of other fixes:
- TTNN All Commit tests were modified during the rebase and I did not notice. Fixed.
- Boolean condition in Docker Run Action should be evaluated as string.

### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: .github/workflows/ttnn-post-commit.yaml [hunk 1]
```diff
@@ -80,12 +80,9 @@ jobs:
         timeout-minutes: ${{ inputs.timeout }}
         uses: ./.github/actions/docker-run
         with:
-          docker_username: ${{ github.actor }}
+          install_wheel: true
```
Comment (dimitri-tenstorrent) [inline .github/workflows/ttnn-post-commit.yaml:83]: Missed this in the code rebase

File: .github/actions/docker-run/action.yml [hunk 2]
```diff
@@ -80,7 +80,7 @@ runs:
           ${{ inputs.device }}
           -w ${{ github.workspace }}
         run: |
-          if [ ${{ inputs.install_wheel }} ]; then
+          if [ ${{ inputs.install_wheel }} = "true" ]; then
```
Comment (dimitri-tenstorrent) [inline .github/actions/docker-run/action.yml:83]: Need to use as a string comparison - the value is not actually boolean. There is an existing Github issue:
- https://github.com/actions/runner/issues/1483

Comment (ttmchiou) [reply .github/actions/docker-run/action.yml:83]: does it matter if its "" or ''?
I didn't realize this was a thing so maybe we should double check if we have other comparisons here too

Comment (dimitri-tenstorrent) [reply .github/actions/docker-run/action.yml:83]: I think both "" and '' evaluate to a string in bash so it does not matter.
Do you think '' is more consistent?

Comment (ttmchiou) [reply .github/actions/docker-run/action.yml:83]: i think as long as we stick to one convention it is fine.

**Review by ttmchiou — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]

**Pull request closed without merging**